src/move/market_metadata/sources/emojicoin_dot_fun_market_metadata.move

```diff
@@ -0,0 +1,624 @@
+        let admins_ref_mut = &mut borrow_registry_mut_checked(admin).admins;
+        let (admin_to_remove_is_admin, admin_to_remove_index) =
+            admins_ref_mut.index_of(&admin_to_remove);
+        assert!(
+            admin_to_remove != @market_metadata,
+            E_ADMIN_TO_REMOVE_IS_MARKET_METADATA_PUBLISHER
+        );
+        assert!(admin_to_remove_is_admin, E_ADMIN_TO_REMOVE_IS_NOT_ADMIN);
+        admins_ref_mut.remove(admin_to_remove_index);
+    }
+
+    public entry fun add_market_properties(
+        admin: &signer,
+        market: address,
+        keys: vector<String>,
+        values: vector<String>
+    ) acquires Registry {
+        assert!(
+            option::is_some(&emojicoin_dot_fun::market_metadata_by_market_address(market)),
+            E_MARKET_NOT_FOUND
+        );
+        assert!(keys.length() == values.length(), E_KEYS_AND_VALUES_NOT_EQUAL_LENGTH);
+        let vault_ref_mut = borrow_registry_mut_checked(admin);
+        let map = &mut vault_ref_mut.map;
+        if (!map.contains(market)) {
+            map.add(market, simple_map::create());
+        };
+        let market_properties = map.borrow_mut(market);
+        while (keys.length() > 0) {
+            market_properties.upsert(keys.pop_back(), values.pop_back());
+        }
+    }
+
+    public entry fun set_market_properties(
+        admin: &signer,
+        market: address,
+        keys: vector<String>,
+        values: vector<String>
+    ) acquires Registry {
+        assert!(
+            option::is_some(&emojicoin_dot_fun::market_metadata_by_market_address(market)),
+            E_MARKET_NOT_FOUND
+        );
+        assert!(keys.length() == values.length(), E_KEYS_AND_VALUES_NOT_EQUAL_LENGTH);
+        let vault_ref_mut = borrow_registry_mut_checked(admin);
+        let map = &mut vault_ref_mut.map;
+        map.upsert(market, simple_map::create());
+        let market_properties = map.borrow_mut(market);
+        while (keys.length() > 0) {
+            market_properties.upsert(keys.pop_back(), values.pop_back());
+        }
+    }
+
+    public entry fun remove_market_properties(
+        admin: &signer, market: address, keys: vector<String>
+    ) acquires Registry {
+        assert!(
+            option::is_some(&emojicoin_dot_fun::market_metadata_by_market_address(market)),
+            E_MARKET_NOT_FOUND
+        );
+        let vault_ref_mut = borrow_registry_mut_checked(admin);
+        let map = &mut vault_ref_mut.map;
+        if (!map.contains(market)) {
+            return;
+        };
+        let market_properties = map.borrow_mut(market);
+        while (keys.length() > 0) {
+            let key = keys.pop_back();
+            if (market_properties.contains_key(&key)) {
+                market_properties.remove(&key);
+            };
+        };
+        if (market_properties.length() == 0) {
+            map.remove(market);
+        };
+    }
+
+    fun init_module(market_metadata: &signer) {
```

Comment:
```suggestion
    /// Remove all properties for a market and the market entry itself, returning silently if the
    /// market entry does not exist.
    public entry fun remove_market_entry(admin: &signer, market: address) acquires Registry {
        assert!(
            option::is_some(&emojicoin_dot_fun::market_metadata_by_market_address(market)),
            E_MARKET_NOT_FOUND
        );
        let vault_ref_mut = borrow_registry_mut_checked(admin);
        let map = &mut vault_ref_mut.map;
        if (map.contains(market)) {
            map.remove(market);
        };
    }

    fun init_module(market_metadata: &signer) {
```

- [ ] Test to 100% coverage
- [ ] Resolve this comment, and all other suggestions once applied

Reply:
This is so that you can remove a market's entire metadata entry by just passing the address, which is not possible per `remove_market_properties`, where you'd have to pass all of its properties

Reply:
You can already call the `set` method with an empty map.

Reply:
But that would still produce an entry in the SmartTable, correct? A map from market address to an empty simple map? That could be misleading, because `remove_market_properties` appears to establish the standard that if a map entry has as its value an empty simple map, then the entry gets wiped

Reply:
You're right. I thought I implemented `set` as I did for `remove` where when you remove everything, it removes the entry. I would rather update `set`, just to keep the code as clean and concise as possible. It also makes sense to have the same behavior in `set` and `remove`. Additionally, on the caller side, this does not introduce a notion of `entry`. Whenever you add something, the `entry` is automatically created, and whenever you remove everything, the `entry` is automatically removed, which is completely transparent for the API consumer. Let me update that.

Reply:
Addressed in 86e2a88028404deea5fcf7a9c024b4fcd2b0b846. See [these lines](https://github.com/econia-labs/emojicoin-dot-fun/blob/86e2a88028404deea5fcf7a9c024b4fcd2b0b846/src/move/market_metadata/sources/emojicoin_dot_fun_market_metadata.move#L118-L121) for implementation and [these lines](https://github.com/econia-labs/emojicoin-dot-fun/blob/86e2a88028404deea5fcf7a9c024b4fcd2b0b846/src/move/market_metadata/sources/emojicoin_dot_fun_market_metadata.move#L362-L391) for tests.